### ISSUE TITLE
playback issue

### ISSUE THREAD

**Issue by bucanero** [irqmask/s3mplay#2]
I don't know if you're still working on `libs3m`, but I was testing around and I found that playing this S3M track is not working as expected: https://github.com/bucanero/apollo-vita/raw/main/data/haiku.s3m

No idea if this `haiku.s3m` file has some special feature that is not supported or makes the playback go wrong, but if you compare and play with another player like https://github.com/libxmp/xmp-cli you'll see that the song is not looping correctly.

**Comment by irqmask** [comment]
Thanks, I'll look into it. This is a little hobby project of mine which I didn't touch for a long time. Especially the effect section is not fully implemented and properly tested and debugged. But I'm glad work work on it, after a long time again :-)

**Comment by irqmask** [comment]
I found several differences and fixed several small bugs, which improved imho playback of haiku.s3m. 
I hope your finding was also fixed.
But don't hesitate to let me know, when you find more glitches

**Comment by bucanero** [comment]
thanks, I'll test it out and share my comments if I found any issues.
I was mostly testing this `haiku.s3m` but I'll also try with other files 👍

**Comment by bucanero** [comment]
hi @irqmask , I was testing a bit more and I can confirm that `haiku.s3m` playback works fine now 👍  

I can also confirm that I was able to build the library for the PlayStation 4 (using the OpenOrbis open-source SDK), and I was able to play the song fine on the PS4 too. 😄 

btw, in case it's useful for your project and improvements, I found this public-domain S3M decoder in C: https://github.com/iamgreaser/f3m/blob/master/f3m.c

**Comment by irqmask** [comment]
Hi @bucanero, 

thanks for testing and I'm glad to hear that it works for you now. :-)
Nice job, porting it to the PS4. :+1:  I only have a PS3, otherwise I'd also give it a try.  Are you planning to use s3mplay in one of your projects?

Btw, Thanks for the link, always interesting how other people solved it.

**Comment by bucanero** [comment]
Yes, I'm planning on using it on one of my ps4 projects [apollo-ps4](https://github.com/bucanero/apollo-ps4/)

I also made some PS3 projects, but on ps3 I used `libxmp` and `mikmod` libraries as those were ported already. But I'm sure that this `libs3m` can be ported to ps3 and other platforms too.

Btw, I added a PS4 [`Makefile`](https://github.com/bucanero/s3mplay/blob/master/Makefile.ps4) on my fork, but I'm not sure if that's something to do a PR or not, as might be out of scope for your project. In any case, I'll ping if I include it in my ps4 project.
